applications/portfolioDrafts/deletePortfolioDraft.ts

```diff
@@ -0,0 +1,56 @@
+import { DeleteCommand, DeleteCommandInput } from "@aws-sdk/lib-dynamodb";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { dynamodbClient as client } from "./utils/dynamodb";
+import { ErrorCodes } from "./models/Error";
+import { ErrorResponse, ErrorStatusCode, NoContentResponse } from "./utils/response";
+
+const TABLE_NAME = process.env.ATAT_TABLE_NAME;
+
+/**
+ * Deletes a Portfolio Draft
+ *
+ * @param event - The DELETE request from API Gateway
+ */
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  const portfolioDraftId = event.pathParameters?.portfolioDraftId;
+
+  if (!portfolioDraftId) {
+    return new ErrorResponse(
+      { code: ErrorCodes.INVALID_INPUT, message: "PortfolioDraftId must be specified in the URL path" },
+      ErrorStatusCode.BAD_REQUEST
+    );
+  }
+
+  if (event.body) {
+    return new ErrorResponse(
+      { code: ErrorCodes.INVALID_INPUT, message: "HTTP request body must be empty" },
+      ErrorStatusCode.BAD_REQUEST
+    );
+  }
+
+  const params: DeleteCommandInput = {
+    TableName: TABLE_NAME,
+    Key: {
+      id: portfolioDraftId,
+    },
+    ReturnValues: "ALL_OLD",
+  };
+
+  const command = new DeleteCommand(params);
+
+  try {
+    const data = await client.send(command);
+    if (!data.Attributes) {
+      return new ErrorResponse(
+        { code: ErrorCodes.INVALID_INPUT, message: "Portfolio Draft with the given ID does not exist" },
+        ErrorStatusCode.NOT_FOUND
+      );
+    }
+    return new NoContentResponse({ message: "Successful operation" });
```

Comment:
Does the constructor require this? I was expecting to work like:

```
return new NoContentResponse()
```

Since there should not be a body and the status code should be 204.

Reply:
As-is, I think this will create a _header_ that looks like:

```
Message: Successful operation
```

That probably won't break anything but I think we should just go with an empty constructor?

Reply:
I realized that I didn't do the delete responses the way I had intended (200 should return the entity being deleted and 204 should return an empty body). We can fix that in another PR though. 

And to Kyle's point - an empty constructor would be preferable to passing boilerplate JSON into the constructor.

Reply:
I was trying to provide the description: Successful operation specified for the 200 in the API, but was just clumsy.  Will remove.

Reply:
Fixed in https://github.com/dod-ccpo/atat-web-api/pull/40/commits/b43d75a429b5d25a08a140d4018707d5828282c7